skaffold-operator.yaml

```diff
@@ -903,3 +905,235 @@ profiles:
+            - pkg/provider-local/imagevector
+            - pkg/provider-local/imagevector/images.yaml
+            - pkg/provider-local/local
+            - pkg/provider-local/webhook/controlplane
+            - pkg/provider-local/webhook/dnsconfig
+            - pkg/provider-local/webhook/networkpolicy
+            - pkg/provider-local/webhook/node
+            - pkg/provider-local/webhook/nodeagentosc
+            - pkg/provider-local/webhook/shoot
+            - pkg/resourcemanager/controller/garbagecollector/references
+            - pkg/utils
+            - pkg/utils/chart
+            - pkg/utils/context
+            - pkg/utils/errors
+            - pkg/utils/flow
+            - pkg/utils/gardener
+            - pkg/utils/gardener/shootstate
+            - pkg/utils/imagevector
+            - pkg/utils/kubernetes
+            - pkg/utils/kubernetes/client
+            - pkg/utils/kubernetes/health
+            - pkg/utils/kubernetes/unstructured
+            - pkg/utils/managedresources
+            - pkg/utils/managedresources/builder
+            - pkg/utils/net
+            - pkg/utils/retry
+            - pkg/utils/secrets
+            - pkg/utils/secrets/manager
+            - pkg/utils/time
+            - pkg/utils/timewindow
+            - pkg/utils/validation/kubernetesversion
+            - pkg/utils/version
+            - third_party/gopkg.in/yaml.v2
+            - VERSION
+        ldflags:
+          - '{{.LD_FLAGS}}'
+        main: ./cmd/gardener-extension-provider-local
+    - image: local-skaffold/gardener-extension-provider-local-node
+      context: pkg/provider-local/node
+      docker: {}
+    - image: local-skaffold/machine-controller-manager-provider-local
+      hooks:
+        after:
+          - command:
+              - bash
+              - hack/mcm-provider-local-operator-overwrite.sh
+      ko:
+        dependencies:
+          paths:
+            - cmd/machine-controller-manager-provider-local
+            - pkg/provider-local/machine-provider/api/v1alpha1
+            - pkg/provider-local/machine-provider/api/validation
+            - pkg/provider-local/machine-provider/local
+            - VERSION
+        ldflags:
+          - '{{.LD_FLAGS}}'
+        main: ./cmd/machine-controller-manager-provider-local
+    - image: local-skaffold/gardener-extension-provider-local/charts/extension
+      custom:
+        buildCommand: |
+          ./hack/push-helm.sh
+        dependencies:
+          paths:
+            - charts/gardener/provider-local
+      requires:
+        - image: local-skaffold/gardener-extension-provider-local
+          alias: IMG
+resourceSelector:
+  allow:
+    # instruct skaffold to inject the built image reference into the image field in our Extension object
+    - groupKind: Extension.operator.gardener.cloud
+      image:
+        - ".spec.deployment.extension.helm.ociRepository.ref"
+manifests:
+  kustomize:
+    paths:
+      - example/provider-local/garden/operator
+deploy:
```

Comment:
Now that `provider-local` is deployed as an `Extension`, we could remove its deployment in `make operator-seed-up` ([ref](https://github.com/gardener/gardener/blob/93bd7a66d019c61eecdbc8bba4b9ce560da40b9b/Makefile#L399)), couldn't we?
A complete removal might not work out of the box, because it also deploys other things ([ref](https://github.com/gardener/gardener/tree/master/example/provider-local/garden/base)), but creating `ControllerRegistrations` and `ControllerDeployments` should not be necessary anymore.